src/tsickle.ts

```diff
@@ -2013,189 +2021,75 @@ export function emitWithTsickle(
+      globalSymbols += `\t\texport {${symbol.name} as module$contents$${clutzModuleName}_${
+          symbol.name}_Instance}\n`;
+      nestedSymbols += `\t\texport {module$contents$${clutzModuleName}_${symbol.name} as ${
+          symbol.name}_Instance}\n`;
     }
   }
 
-  if (reexportsByNamespace.size) {
-    dtsFileContent += 'declare global {\n';
-    for (const [namespace, rexps] of reexportsByNamespace) {
-      dtsFileContent += `\tnamespace ${namespace} {\n`;
-      for (const rexp of rexps) {
-        dtsFileContent += `\t\t${rexp}\n`;
-      }
-      dtsFileContent += '\t}\n';
-    }
-    dtsFileContent += '}\n';
-  }
+  dtsFileContent += 'declare global {\n';
+  dtsFileContent += `\tnamespace ಠ_ಠ.clutz {\n`;
+  dtsFileContent += globalSymbols;
+  dtsFileContent += `\t}\n`;
+  dtsFileContent += `\tnamespace ಠ_ಠ.clutz.module$exports$${clutzModuleName} {\n`;
+  dtsFileContent += nestedSymbols;
+  dtsFileContent += `\t}\n`;
+  dtsFileContent += '}\n';
 
   return dtsFileContent;
 }
 
-/**
- * Returns 2 strings specifying the generic type arguments for the symbol.  The constrained params
- * include any `T extends foo` arguments, the regular params are just a list of the type symbols,
- * since we need the constraints on the LHS of the alias declaration, but can't have them on the
- * RHS.
- */
-function getGenericTypeParameters(symbol: ts.Symbol):
-    {params: string, paramsWithContraint: string} {
-  if (!symbol.declarations) {
-    return {params: '', paramsWithContraint: ''};
-  }
-
-  // All declarations have to have matching generic types, so we're safe just looking at
-  // the first one.
-  if (!symbol.declarations[0]) {
-    return {params: '', paramsWithContraint: ''};
-  }
-
-  const declaration = symbol.declarations[0];
-
-  if ([
-        ts.SyntaxKind.FunctionDeclaration, ts.SyntaxKind.ConstructorKeyword,
-        ts.SyntaxKind.ClassDeclaration, ts.SyntaxKind.InterfaceDeclaration,
-        ts.SyntaxKind.TypeAliasDeclaration
-      ].indexOf(declaration.kind) === -1) {
-    return {params: '', paramsWithContraint: ''};
-  }
-
-  const declarationWithTypeParameters: ts.DeclarationWithTypeParameters =
-      declaration as ts.DeclarationWithTypeParameters;
-
-  if (!declarationWithTypeParameters.typeParameters) {
-    return {params: '', paramsWithContraint: ''};
-  }
-
-  const paramList: string[] = [];
-  const constrainedParamList: string[] = [];
-  for (const param of declarationWithTypeParameters.typeParameters) {
-    let constrainedParam = param.name.getText();
-    if (param.constraint) {
-      constrainedParam += ` extends ${param.constraint.getText()}`;
-    }
-    if (param.default) {
-      constrainedParam += ` = ${param.default.getText()}`;
-    }
-    constrainedParamList.push(constrainedParam);
-    paramList.push(param.name.getText());
-  }
-
-  const params = `<${paramList.join(',')}>`;
-  const paramsWithContraint = `<${constrainedParamList.join(',')}>`;
-
-  return {params, paramsWithContraint};
-}
```

Comment:
Overall this is a lot of deleted code (good!) but it probably would be useful to have the author of this code (Lucas? Rado?) take a look too.

Reply:
@LucasSloan PTAL.